scripts/py_matter_idl/matter_idl/data_model_xml/handlers/context.py

```diff
@@ -82,6 +82,19 @@ def __init__(self, locator: Optional[xml.sax.xmlreader.Locator] = None):
         self.file_name = None
         self._not_handled: set[str] = set()
         self._idl_post_processors: list[IdlPostProcessor] = []
+        self.base_clusters: dict[str, Cluster] = {}
+
+    def AddBaseCluster(self, name: str, parse_meta: Optional[ParseMetaData] = None) -> Cluster:
+        """Creates a new cluster entry for the given name in the list of known
+           base clusters.
+        """
+        assert name not in self.base_clusters  # be unique
+
+        cluster = Cluster(side=ClusterSide.CLIENT, name=name,
```

Comment:
What about ClusterSide.SERVER? Why are base clusters listed as "client" side?

Reply:
I believe trying server/client was a design/understanding mistake in matter.

Now in practical terms, what I had was:
  - zap differentiates between server clusters and client clusters (bindings) and has different loops for them
  - originally I thought that we would have "client cluster" == "a cluster with all enabled", because may want anything on client side and link-time optimization would strip out unused code. Then "server cluster" == exposed bits by the server.
  
  In the mean time I learned that server-cluster may be different across endpoints, so the reality we have is that endpoints now expose what attributes, commands (and events if applicable) it has, so having basically duplicate (but slightly different) server and client clusters is quite silly. We do it just because zap does it and we need to be able to generate 100% identical output from zapt and jinja templates  ... all this is basically legacy.
  
Now specifically for these parsers and diffs: I am using the parsers to compare against the 'client clusters' matter file, which is a 100% unfiltered view of clusters, tagging them "client". This completely ignores any zap.

Hence this "client" bit. There should be no client bit, however making that change would be quite large and independent of this. Most of the differ does not actually care about client/server at all (especially for base clusters) however we have to pick one and text diff outputs the type in text ... so we use client to make the diff match.